script/deployPoolWithRenzoStabilityAndMintLiquidity.s.sol

```diff
@@ -0,0 +1,304 @@
+        PoolKey memory pool = PoolKey({
+            currency0: currency0,
+            currency1: currency1,
+            fee: lpFee,
+            tickSpacing: tickSpacing,
+            hooks: hookContract
+        });
+        bytes memory hookData = new bytes(0);
+
+        // --------------------------------- //
+        // get tick range
+        (tickLower, tickUpper) = _getTickRange(pool, startingPrice);
+        console2.log("Lower tick for liquidity -> ", tickLower);
+        console2.log("Upper tick for liquidity -> ", tickUpper);
+
+        // Converts token amounts to liquidity units
+        uint128 liquidity = LiquidityAmounts.getLiquidityForAmounts(
+            startingPrice,
+            TickMath.getSqrtPriceAtTick(tickLower),
+            TickMath.getSqrtPriceAtTick(tickUpper),
+            token0Amount,
+            token1Amount
+        );
+
+        // slippage limits
+        uint256 amount0Max = token0Amount + 1 wei;
+        uint256 amount1Max = token1Amount + 1 wei;
+
+        (
+            bytes memory actions,
+            bytes[] memory mintParams
+        ) = _mintLiquidityParams(
+                pool,
+                tickLower,
+                tickUpper,
+                liquidity,
+                amount0Max,
+                amount1Max,
+                recipient,
+                hookData
+            );
+
+        // multicall parameters
+        bytes[] memory params = new bytes[](2);
+
+        // initialize pool
+        params[0] = abi.encodeWithSelector(
+            posm.initializePool.selector,
+            pool,
+            startingPrice
+        );
+
+        // mint liquidity
+        params[1] = abi.encodeWithSelector(
+            posm.modifyLiquidities.selector,
+            abi.encode(actions, mintParams),
+            block.timestamp + 60
+        );
+
+        // if the pool is an ETH pair, native tokens are to be transferred
+        uint256 valueToPass = currency0.isAddressZero() ? amount0Max : 0;
+
+        vm.startBroadcast();
+        tokenApprovals(amount0Max, amount1Max);
+        vm.stopBroadcast();
+
+        // multicall to atomically create pool & add liquidity
+        vm.startBroadcast();
+        posm.multicall{value: valueToPass}(params);
+        vm.stopBroadcast();
+
+        console2.log("poolId of the deployed pool - ");
+        console2.logBytes32(PoolId.unwrap(pool.toId()));
+    }
+
+    function _deployHook() internal returns (address) {
+        // hook contracts must have specific flags encoded in the address
+        uint160 flags = uint160(
+            Hooks.AFTER_INITIALIZE_FLAG | Hooks.BEFORE_SWAP_FLAG
+        );
+
+        // Mine a salt that will produce a hook address with the correct flags
+        bytes memory constructorArgs = abi.encode(
+            POOLMANAGER,
+            rateProvider,
+            minFee,
+            maxFee,
+            ezETH
+        );
+        (address hookAddress, bytes32 salt) = HookMiner.find(
+            CREATE2_FACTORY,
```

Comment:
not sure where this constant is coming from
```suggestion
            CREATE2_DEPLOYER,
```

Reply:
used the one from foundry lib itself. Updated to `CREATE2_DEPLOYER` 🫡